pkg/cephmgr/osd/daemon.go

```diff
@@ -90,9 +104,56 @@ func getAvailableDevices(context *clusterd.DaemonContext, devices []*inventory.L
 			return nil, fmt.Errorf("failed to get device %s info. %+v", device.Name, err)
 		}
 		if fs == "" && ownPartitions {
-			available.Entries[device.Name] = &DeviceOsdIDEntry{Data: unassignedOSDID}
+			if desiredDevices == "all" {
+				available.Entries[device.Name] = &DeviceOsdIDEntry{Data: unassignedOSDID}
+			} else {
+				logger.Infof("skipping device %s until the admin specifies it can be used by an osd", device.Name)
+			}
 		}
 	}
 
 	return available, nil
 }
+
+func getDataDirs(context *clusterd.DaemonContext, useDataDir bool) (map[string]int, error) {
```

Comment:
i don't see this being called with false anywhere but from test code.  will it be in the future?